addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/handler/FeedHandler.java

```diff
@@ -0,0 +1,290 @@
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.io.FeedException;
+import com.rometools.rome.io.SyndFeedInput;
+
+/**
+ * The {@link FeedHandler } is responsible for handling commands, which are
+ * sent to one of the channels and for the regular updates of the feed data.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class FeedHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(FeedHandler.class);
+
+    private String urlString;
+    private BigDecimal refreshTime;
+    private ScheduledFuture<?> refreshTask;
+    private SyndFeed currentFeedState;
+
+    public FeedHandler(Thing thing) {
+        super(thing);
+        currentFeedState = null;
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        super.initialize();
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        logger.debug("Start reading Feed Thing configuration.");
+        Configuration configuration = getConfig();
+
+        // It is not necessary to check if the URL is valid, this will be done in fetchFeedData() method
+        urlString = (String) configuration.get(URL);
+
+        try {
+            refreshTime = (BigDecimal) configuration.get(REFRESH_TIME);
+            if (refreshTime.intValue() <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}. {}", refreshTime,
+                    DEFAULT_REFRESH_TIME, e.getMessage());
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+
+    }
+
+    private void startAutomaticRefresh() {
+
+        Runnable refresher = new Runnable() {
```

Comment:
This method is called on every channelLinked() call, of which there can be many.
You must check that there isn't already a refresh job scheduled on subsequent calls.